src/Modules/SimplCommerce.Module.Core/Services/SettingService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using SimplCommerce.Infrastructure.Extensions;
+using SimplCommerce.Infrastructure.Helpers;
+using SimplCommerce.Infrastructure.Models;
+using SimplCommerce.Module.Core.Extensions;
+using SimplCommerce.Module.Core.Models;
+
+namespace SimplCommerce.Module.Core.Services
+{
+    public class SettingService : ISettingService
+    {
+        private readonly UserManager<User> _userManager;
+        private readonly SettingDefinitionProvider _settingDefinitionProvider;
+        private readonly User _currentUser;
+
+        public SettingService(
+            UserManager<User> userManager,
+            SettingDefinitionProvider settingDefinitionProvider,
+            IWorkContext workContext)
+        {
+            _userManager = userManager;
+            _settingDefinitionProvider = settingDefinitionProvider;
+            _currentUser = AsyncHelper.RunSync(workContext.GetCurrentUser);
```

Comment:
Sorry for the late response, but I think we should remove this line from constructor.  Individual method that need currentUser will call to get it so that we can async all the way and don't have to use AsyncHelper

Reply:
Thanks reply, I will refactor it.